packages/wouter/src/memory-location.js

```diff
@@ -7,10 +7,18 @@ import { useSyncExternalStore } from "./react-deps.js";
 
 export const memoryLocation = ({
   path = "/",
+  searchPath = "",
   static: staticLocation,
   record,
 } = {}) => {
-  let currentPath = path;
+  let initialPath = path;
+  if (searchPath) {
+    // join with & if path contains search query, and ? otherwise
+    initialPath += path.split("?")[1] ? "&" : "?";
+    initialPath += searchPath;
+  }
+
+  let [currentPath, currentSearch = ""] = initialPath.split("?");
   const history = [currentPath];
```

Comment:
Just one more thing to fix the initial entry in the history array, now that `currentPath` doesn’t include the query string anymore.

```suggestion
  const history = [initialPath];
```

Otherwise this wouldn’t work:

```js
const { history } = memoryLocation({ path: "/foo?key=value" });
// history[0] is `/foo`, but should be `/foo?key=value`
```